hibernate-core/src/main/java/org/hibernate/boot/model/source/spi/PluralAttributeMapKeySource.java

```diff
@@ -39,7 +39,7 @@ enum Nature {
 	 * (relevant only for one-to-many and many-to-many associations)?
 	 *
 	 * If this method returns {@code true}, then this object can safely
-	 * be cast to {@link PluralAttributeMapKeySourceEntityAttribute}.
+	 * be cast to {@link PluralAttributeMapKeyManyToAnySource}.
 	 *
 	 * @return true, if this plural attribute index source for an attribute of the referenced
 	 * entity; false, otherwise.
```

Comment:
The above change is the most difficult one and I am not sure it is correct. Let me know if the change is erroneous.